### PR TITLE
Replace reduce(true) and reduce(false) with allSatisfy

### PR DESCRIPTION
**Pull request by chriseidhof**
I replaced all `reduce(true, { $0 && $1 })` with `allSatisfy { $0 }`'s. (Likewise for `false` and `||`).

### Motivation:

https://twitter.com/Lukasaoz/status/1024691134885453824

### Modifications:

See above.

### Result:

Whether it's clearer is always up to the reader, but I think this PR makes good use of the standard library. Ideally, we'd have something like allSatisfy at the XCTest level (for better error messages). But I just wanted to get my PR in.

### CODE REVIEW

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by chriseidhof** [comment]
Ah, of course I didn't run the tests, spotted a tiny mistake. That should be fixed now.

**Comment by weissi** [comment]
@swift-nio-bot test this please

File: Tests/NIOHTTP1Tests/HTTPResponseCompressorTest.swift [hunk 1]
```diff
@@ -39,11 +39,11 @@ private class PromiseOrderer {
         promise.futureResult.whenComplete {
             let priorFutures = self.promiseArray[0..<thisPromiseIndex]
             let subsequentFutures = self.promiseArray[(thisPromiseIndex + 1)...]
-            let allPriorFuturesFired = priorFutures.map { $0.futureResult.isFulfilled }.reduce(true, { $0 && $1 })
-            let allSubsequentFuturesUnfired = subsequentFutures.map { $0.futureResult.isFulfilled }.reduce(false, { $0 || $1 })
+            let allPriorFuturesFired = priorFutures.map { $0.futureResult.isFulfilled }.allSatisfy { $0 }
```
Comment (ephemer) [inline Tests/NIOHTTP1Tests/HTTPResponseCompressorTest.swift]: I think all of these use cases would be clearer, and probably more performant, without the initial map:

```suggestion
            let allPriorFuturesFired = priorFutures.allSatisfy { $0.futureResult.isFulfilled }
```

File: Tests/NIOHTTP1Tests/HTTPResponseCompressorTest.swift [hunk 2]
```diff
@@ -39,11 +39,11 @@ private class PromiseOrderer {
...
+            let allSubsequentFuturesUnfired = subsequentFutures.map { $0.futureResult.isFulfilled }.allSatisfy { !$0 }
```
Comment (ephemer) [inline Tests/NIOHTTP1Tests/HTTPResponseCompressorTest.swift]:
```suggestion
            let allSubsequentFuturesUnfired = subsequentFutures.allSatisfy { !$0.futureResult.isFulfilled }
```

File: Tests/NIOTests/ChannelTests.swift [hunk 3]
```diff
@@ -669,7 +669,7 @@ public class ChannelTests: XCTestCase {
 
             pwm.failAll(error: ChannelError.operationUnsupported, close: true)
 
-            XCTAssertTrue(ps.map { $0.futureResult.isFulfilled }.reduce(true) { $0 && $1 })
+            XCTAssertTrue(ps.map { $0.futureResult.isFulfilled }.allSatisfy { $0 })
```
Comment (ephemer) [inline Tests/NIOTests/ChannelTests.swift]:
```suggestion
            XCTAssertTrue(ps.allSatisfy { $0.futureResult.isFulfilled })
```

File: Tests/NIOTests/DatagramChannelTests.swift [hunk 4]
```diff
@@ -192,7 +192,7 @@ final class DatagramChannelTests: XCTestCase {
 
         // Now close the channel. When that completes, all the futures should be complete too.
         let fulfilled = try self.firstChannel.close().map {
-            promises.map { $0.isFulfilled }.reduce(true, { $0 && $1 })
+            promises.map { $0.isFulfilled }.allSatisfy { $0 }
```
Comment (ephemer) [inline Tests/NIOTests/DatagramChannelTests.swift]:
```suggestion
            promises.allSatisfy { $0.isFulfilled }
```

File: Tests/NIOTests/PendingDatagramWritesManagerTests.swift [hunk 5]
```diff
@@ -426,7 +426,7 @@ class PendingDatagramWritesManagerTests: XCTestCase {
 
             pwm.failAll(error: ChannelError.operationUnsupported, close: true)
 
-            XCTAssertTrue(ps.map { $0.futureResult.isFulfilled }.reduce(true) { $0 && $1 })
+            XCTAssertTrue(ps.map { $0.futureResult.isFulfilled }.allSatisfy { $0 })
```
Comment (ephemer) [inline Tests/NIOTests/PendingDatagramWritesManagerTests.swift]:
```suggestion
            XCTAssertTrue(ps.allSatisfy { $0.futureResult.isFulfilled })
```

**Comment by weissi** [comment]
@chriseidhof / @ephemer I just re-tagged this for 2.0.0 which will be Swift 5+ so we can just use the new `allSatisfy` :)

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Pull request merged**